src/scope.ml

```diff
@@ -118,6 +118,95 @@ type mode =
   (** Right-hand side mode. The constructor carries the environment for higher
       order variables that will be bound in the representation of the RHS. *)
 
+(* splitFunArgs t decomposes the parser term [t] (which is expected to be an application),
+  into the function symbol and a list of its actual arguments. It is implemented 
+  as a tail-recursive function. *)
+let splitFunArgs : p_term -> (p_term * (p_term list)) = fun t ->
+  let rec splitFunArgs_aux : p_term -> (p_term list) -> (p_term * (p_term list)) = fun t args ->
+    match t.elt with
+    | P_Appl(u,v) -> splitFunArgs_aux u (v::args)
+    | _           -> (t, args)
+  in
+  splitFunArgs_aux t []
+
+(* Just for debugging *)
+let boolList_to_string : bool list -> string = fun l ->
```

Comment:
Should be in extra.ml.

Reply:
It's just a function for debugging that has no interest. I'll remove it.